commit 6d9c153a0b84392406bc77600aa7d3ea365de041
Author: Florian Fainelli <f.fainelli@gmail.com>
Date:   Sat Sep 2 08:56:45 2017 -0700

    net: dsa: loop: Do not unregister invalid fixed PHY
    
    During error injection it was possible to crash in dsa_loop_exit() because of
    an attempt to unregister an invalid PHY. We actually want to the driver probing
    in dsa_loop_init() even though fixed_phy_register() may return an error to
    exercise how DSA deals with such cases, but we should not be crashing during
    driver removal.
    
    Fixes: 98cd1552ea27 ("net: dsa: Mock-up driver")
    Signed-off-by: Florian Fainelli <f.fainelli@gmail.com>
    Reviewed-by: Andrew Lunn <andrew@lunn.ch>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/dsa/dsa_loop.c b/drivers/net/dsa/dsa_loop.c
index fdd8f3872102..8150e3a3c8dd 100644
--- a/drivers/net/dsa/dsa_loop.c
+++ b/drivers/net/dsa/dsa_loop.c
@@ -390,7 +390,7 @@ static void __exit dsa_loop_exit(void)
 
 	mdio_driver_unregister(&dsa_loop_drv);
 	for (i = 0; i < NUM_FIXED_PHYS; i++)
-		if (phydevs[i])
+		if (!IS_ERR(phydevs[i]))
 			fixed_phy_unregister(phydevs[i]);
 }
 module_exit(dsa_loop_exit);